src/core/annotation.js

```diff
@@ -1359,26 +1369,41 @@ class WidgetAnnotation extends Annotation {
     return [initialState.font, initialState.fontName, initialState.fontSize];
   }
 
-  _computeFontSize(font, fontName, fontSize, height) {
+  _computeFontSize(font, fontName, fontSize, height, lineCount) {
     if (fontSize === null || fontSize === 0) {
-      const em = font.charsToGlyphs("M")[0].width / 1000;
-      // According to https://en.wikipedia.org/wiki/Em_(typography)
-      // an average cap height should be 70% of 1em
-      const capHeight = 0.7 * em;
-      // 1.5 * capHeight * fontSize seems to be a good value for lineHeight
-      fontSize = Math.max(1, Math.floor(height / (1.5 * capHeight)));
+      if (lineCount === -1) {
+        fontSize = Math.round(0.8 * height * 10) / 10;
+      } else {
+        // Hard to guess how many lines there are
```

Comment:
I don't understand this comment: we know exactly how many lines there are, namely `lineCount`?

Reply:
The problem here is that if the textarea is "designed" to have 3 lines and the user enters text on 1 line then the font will be too big.
It's exactly what happens in the example gave by the user in the bug.